gam/gam.py

```diff
@@ -114,44 +128,48 @@ def _get_explanations(self, centers):
             explanations.append(list(zip(self.feature_labels, explanation_weights)))
         return explanations
 
-    def get_optimal_clustering(self, cluster_list=[2, 3, 4, 5, 6]):
-        from sklearn.metrics import silhouette_score  # silhouette_samples,
-        from gam.spearman_distance import pairwise_spearman_distance_matrix
-        from gam.kendall_tau_distance import pairwise_distance_matrix
+    def get_optimal_clustering(self, max_clusters=2, verbose=False):
+        """Automatically select optimal cluster count
+
+        Args:
+            cluster (int): maximum amount of clusters to test
+
+        Returns: None
+        """
+        silh_list = []
+        max_clusters = max(2,max_clusters)
 
-        silhList = []
-        for nCluster in cluster_list:
-            self.k = nCluster
+        for n_cluster in range(2, max_clusters + 1):
+            self.k = (
+                n_cluster
+            )  # Updating the class attribute repeatedly may not be best practice
             self.generate()
 
             # TODO - save GAM clusters to pkl file - saves recomputing
-            if self.distance == 'spearman':
+            if self.distance == "spearman":
```

Comment:
Do you prefer `"` over `'` ? Is this a black formatting thing? Another thought for linting standards

Reply:
It's a black formatting thing here, but yeah I tend to use `""`